### ISSUE TITLE
Missing package: cryptography

### ISSUE THREAD

**Issue by bennuttall** [piwheels/packages#464]
### Package name

cryptography

### Package version

43.0.0

### PyPI URL

https://pypi.org/project/cryptography/

### piwheels URL

https://www.piwheels.org/project/cryptography/

### Python version

- [X] Python 3.9
- [X] Python 3.11

### I am aware this is the issue tracker for a Python package index specifically for Raspberry Pi

- [X] Yes

### I have checked for duplicate issues

- [X] Yes

### I am the maintainer

- [ ] Yes

### More information

Issues arose in #462 and #463

cryptography now requires openssl 3, so we need to statically link it in Bullseye which has openssl 1.1.

The recommendation was to build it with `OPENSSL_STATIC=1` set but that apparently didn't work.

**Comment by bennuttall** [comment]
Figuring out how to fix it, but for now, please download these files and install as required:

- Bullseye (Python 3.9): https://www.piwheels.org/cp39/cryptography-43.0.0-cp37-abi3-linux_armv7l.whl
- Bookworm (Python 3.11): https://www.piwheels.org/cp311/cryptography-43.0.0-cp37-abi3-linux_armv7l.whl

**Comment by orosam** [comment]
Somewhat updating my question in #469: 

Wouldn't  fixing the tags in the build process with auditwheel to something like `cp37-abi3-manylinux_2_31_armv7l` for bullseye, and `cp37-abi3-manylinux_2_36_armv7l` for bookworm differentiate the built wheels sufficiently?

**Comment by xmatthias** [comment]
seems like wheels for 43.0.3 are available now ...


![image](https://github.com/user-attachments/assets/a52885ed-eb56-4d3f-b011-52e3aa891874)

is this a fluke, or has this process been fixed?  (i didn't test them yet).

**Comment by orosam** [comment]
I just performed a quick test.

The 43.0.3 armv7l build (having `sha256=aa694873d3f46d2af4480573597b341f891acdeefe21f37b9112df7004e4cd3e`) appears to be linked against openssl3, and indeed it works on a bookworm based OS.

This coincides with the dependency info in the Project JSON at <https://www.piwheels.org/project/cryptography/json/>, under key `.["releases"]["43.0.3"]["files"]["cryptography-43.0.3-cp37-abi3-linux_armv7l.whl"]["apt_dependencies"]`.

Whether it's deliberate or accidental that the bookworm wheel is there, I cannot tell. Previously (around 43.0.1 or 43.0.2 if I recall correctly), it was the same situation, just reversed: the bookworm wheel was missing, and the bullseye one was available.

**Comment by xmatthias** [comment]
i'm not sure if this is still necessary.
cryptography now (since 44.0.0) provides armv7 wheels via pypi - and a test showed that they work fine on 3.11 (so bookworm).

**Comment by alex** [comment]
Yes, those are abi3 wheels, so the Python version there is the minimum version.

**Comment by MichaIng** [comment]
> i'm not sure if this (building cryptography as part of piwheels) is still necessary (not sure what the policies are for packages that provide upstream wheels).

Those wheels do not work on ARMv6 RPi models, i.e. RPi 1 and Zero.

**Comment by MichaIng** [comment]
Let's close this in favor of a recent report: #619
